File: src/dotnet/PolyBus/Transport/Transactions/IncomingTransaction.cs
using System.Diagnostics;
using PolyBus.Transport.Transactions.Messages;

namespace PolyBus.Transport.Transactions;

[DebuggerStepThrough]
public class IncomingTransaction(IPolyBus bus, IncomingMessage incomingMessage) : Transaction(bus)
{
    /// <summary>
    /// The incoming message from the transport being processed.
    /// </summary>
    public virtual IncomingMessage IncomingMessage { get; set; } = incomingMessage ?? throw new ArgumentNullException(nameof(incomingMessage));
}

File: src/kotlin/src/main/kotlin/polybus/transport/transactions/IncomingTransaction.kt
package polybus.transport.transactions

import polybus.IPolyBus
import polybus.transport.transactions.messages.IncomingMessage

open class IncomingTransaction(bus: IPolyBus, incomingMessage: IncomingMessage) : Transaction(bus) {
    open var incomingMessage: IncomingMessage = incomingMessage
}
